types/jasmine/index.d.ts

```diff
@@ -272,6 +278,7 @@ declare namespace jasmine {
     function addCustomEqualityTester(equalityTester: CustomEqualityTester): void;
 
     function addMatchers(matchers: CustomMatcherFactories): void;
+    function addAsyncMatchers(matchers: CustomMatcherFactories): void;
```

Comment:
There is also the async matcher `toBeRejectedWithError` that has been added in 3.5.0 (see [async-matchers](https://jasmine.github.io/api/3.5/async-matchers.html). Please add it to the AsyncMatchers

Reply:
Hi @djungowski do you have any tips for the matchers? I'm having some issues with the tests. They don't seem to accept the Error parameter.

The commit with the change is here: https://github.com/simondel/DefinitelyTyped/commit/ed90db146791e6bc72db73ef14975c82dd194739

After It's fixed, I'll create a new PR.

Reply:
@simondel From what I see, a test like
```
await expectAsync(Promise.reject(badness)).toBeRejectedWithError(Error, "badness");
```
would have to look like
```
await expectAsync(Promise.reject(new Error('badness'))).toBeRejectedWithError(new Error('badness'));
```

Reply:
@simondel If `toBeRejectedWithError` works like `toThrowError`, it might also work with

```
await expectAsync(Promise.reject(new Error('badness')).toBeRejectedWithError('badness');
```

Reply:
Yeah, I was thinking about that as well.. But that's not what the tests inside of Jasmine use, so I'm not sure if that's the test that I should have in the types.

If you look at lines 5 through 8, or 19 through 22 [of the PR](https://github.com/jasmine/jasmine/pull/1686/files#diff-ab83340b986e6941a2b629c15a1ae79f) you can see code like this:
```js
var matcher = jasmineUnderTest.asyncMatchers.toBeRejectedWithError(jasmineUnderTest.matchersUtil),
actual = Promise.reject(new TypeError('foo'));

matcher.compare(actual, TypeError) // TypeError is used instead of a new instance
```
and
```js
matcher = jasmineUnderTest.asyncMatchers.toBeRejectedWithError(jasmineUnderTest.matchersUtil),
actual = Promise.reject(new TypeError('foo'));

matcher.compare(actual, TypeError, 'foo') // TypeError is used instead of a new instance
```

Reply:
So I've looked at the typing definition of `toThrow` and the tests work if you change your new typings for `toBeRejectedWithError` as follows:

index.d.ts
```javascript
toBeRejectedWithError(message?: string | RegExp): PromiseLike<void>;
toBeRejectedWithError(expected?: new (...args: any[]) => Error, message?: string | RegExp): PromiseLike<void>;
```

ts3.1/index.d.ts
```javascript
toBeRejectedWithError(message?: string | RegExp): Promise<void>;
toBeRejectedWithError(expected?: new (...args: any[]) => Error, message?: string | RegExp): Promise<void>;
```

(The only difference between index.d.ts and ts3.1/index.d.ts is PromiseLike vs. Promise)

Reply:
Thanks for the help! ♥ I've created a new PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/40568